File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        
        r, l = 0, len(heights) - 1
        max_water = 0
        while r < l:
            distance = l - r
            water = min(heights[r], heights[l]) * distance
            max_water = max(max_water, water)

            if heights[r] <= heights[l]:
                r += 1
            else:
                l -= 1

        return max_water

File: Data Structures & Algorithms/jump-game/submission-3.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        
        int farthestJump = 0;
        for (int i = 0; i < nums.size(); i++){
            if ( farthestJump >= i ){
                farthestJump = std::max(farthestJump, i + nums[i]);
            }
            else {
                return false;
            }
        }
        return true;
        
    }
};
